WASI doesn't have syslog().  This has been fixed in upstream OpenSSL-3.2.0+.
https://github.com/openssl/openssl/commit/d88a0f5f3944535dd83d55bedc4d239544c5678d

--- a/crypto/bio/bss_log.c
+++ b/crypto/bio/bss_log.c
@@ -24,6 +24,8 @@
 
 #if defined(OPENSSL_SYS_WINCE)
 #elif defined(OPENSSL_SYS_WIN32)
+#elif defined(__wasi__)
+# define NO_SYSLOG
 #elif defined(OPENSSL_SYS_VMS)
 # include <opcdef.h>
 # include <descrip.h>
